### ISSUE TITLE
Timeout of 15000ms exceeded when switching from Home to other pages

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#2112]
### Expected Behavior

the issue we identified with Brian: on the homepage at the bottom, there's an element that logs errors and it's not properly unloaded when switching pages from Home to any other page - flows/logs etc.

![image](https://github.com/kestra-io/kestra/assets/86264395/51f6f3bf-9d8b-4366-9ac4-c32552d42603)

### Environment Information

- Kestra Version: 0.12 snapshot develop full image with Docker Compose

**Comment by anna-geller** [comment]
issue doesn't happen anymore on my local instance; we only found it on the preview when loading logs

![image](https://github.com/kestra-io/kestra/assets/86264395/aec3b807-ada7-4c80-a5e2-ebabd11f024b)


TODO: improve the DB index or optimize the log query for the Logs page -- it's likely only a Postgres indexing issue, not a UI issue

**Comment by loicmathieu** [comment]
On the logs page, by default, there is no date filter which is causing the full table to be read. We cannot really do anything about it, we should set a default date filter (like 30 days as in the homepage).

**Comment by loicmathieu** [comment]
@anna-geller the same issue exist on the executions page, I think we should set a default date filter of 30 days on both these pages as otherwise they'll read the full table then sort on the date which is impossible for large tables.

**Comment by anna-geller** [comment]
@loicmathieu good point - WDYT about only 7 days filter by default? seems like a good option for those who run thousands of executions

**Comment by tchiotludo** [comment]
even if there is a limit ?

**Comment by loicmathieu** [comment]
@tchiotludo in fact there should be a limit as there is pagination, I'll double-check what was done on the log page.

**Comment by loicmathieu** [comment]
I checked and the LogRepository didn't fetch a page like other repository but all logs. That's why it takes so long.
So the bug is that the JDBC repository didn't paginate the find query.

**Comment by loicmathieu** [comment]
Forget my last comment I was on 0.9 checking for a bug so not looking at the latest code that correctly select a page.
As we order by timestamp, even if we only load 50 logs we must scan the all table, if we add a time we will only scan the index so it will work (and the same issue should exist for the execution page).

So back to my idea of limiting to 30 (or 7) days by default on both the executions and logs page.

**Comment by tchiotludo** [comment]
> As we order by timestamp, even if we only load 50 logs we must scan the all table, if we add a time we will only scan the index so it will work (and the same issue should exist for the execution page).
Index on the timestamp must limit on the latest 50 only, it's the main goal of this index

**Comment by loicmathieu** [comment]
Postgres is not a vector database, indexes are stored using a B-Tree structure, which is not suitable for limiting the number of rows scanned so all rows will be scanned (whether on the index or on the table which will not make any difference).
